src/components/KeyboardSpacer/BaseKeyboardSpacer.js

```diff
@@ -1,69 +1,51 @@
-import React, {PureComponent} from 'react';
+import React, {useState, useEffect, useCallback} from 'react';
 import {Dimensions, Keyboard, View} from 'react-native';
 import * as StyleUtils from '../../styles/StyleUtils';
 import {propTypes, defaultProps} from './BaseKeyboardSpacerPropTypes';
 
-class BaseKeyboardSpacer extends PureComponent {
-    constructor(props) {
-        super(props);
-        this.state = {
-            keyboardSpace: 0,
-        };
-        this.keyboardListeners = null;
-        this.updateKeyboardSpace = this.updateKeyboardSpace.bind(this);
-        this.resetKeyboardSpace = this.resetKeyboardSpace.bind(this);
-    }
-
-    componentDidMount() {
-        const updateListener = this.props.keyboardShowMethod;
-        const resetListener = this.props.keyboardHideMethod;
-        this.keyboardListeners = [Keyboard.addListener(updateListener, this.updateKeyboardSpace), Keyboard.addListener(resetListener, this.resetKeyboardSpace)];
-    }
-
-    componentWillUnmount() {
-        this.keyboardListeners.forEach((listener) => listener.remove());
-    }
+function BaseKeyboardSpacer(props) {
+    const [keyboardSpace, setKeyboardSpace] = useState(0);
 
     /**
      * Update the height of Keyboard View.
      *
      * @param {Object} [event] - A Keyboard Event.
      */
-    updateKeyboardSpace(event) {
+    const updateKeyboardSpace = useCallback((event) => {
         if (!event.endCoordinates) {
             return;
         }
 
         const screenHeight = Dimensions.get('window').height;
-        const keyboardSpace = screenHeight - event.endCoordinates.screenY + this.props.topSpacing;
-        this.setState(
-            {
-                keyboardSpace,
-            },
-            this.props.onToggle(true, keyboardSpace),
-        );
-    }
+        const space = screenHeight - event.endCoordinates.screenY + props.topSpacing;
+        setKeyboardSpace(space);
+        props.onToggle(true, keyboardSpace);
+    }, [keyboardSpace, props]);
 
     /**
      * Reset the height of Keyboard View.
      *
      * @param {Object} [event] - A Keyboard Event.
      */
-    resetKeyboardSpace() {
-        this.setState(
-            {
-                keyboardSpace: 0,
-            },
-            this.props.onToggle(false, 0),
-        );
-    }
+    const resetKeyboardSpace = useCallback(() => {
+        setKeyboardSpace(0);
+        props.onToggle(false, 0);
+    }, [setKeyboardSpace, props]);
+
+    useEffect(() => {
+        const updateListener = props.keyboardShowMethod;
+        const resetListener = props.keyboardHideMethod;
+        const keyboardListeners = [Keyboard.addListener(updateListener, updateKeyboardSpace), Keyboard.addListener(resetListener, resetKeyboardSpace)];
+
+        return () => {
+            keyboardListeners.forEach((listener) => listener.remove());
+        }
+    }, [props.keyboardShowMethod, props.keyboardHideMethod, updateKeyboardSpace, resetKeyboardSpace]);
```

Comment:
We only want to have this side effect on the mount. With passing the dependency, we will have several times call this.

```suggestion
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
```

Reply:
Resolved